src/Type.cpp

```diff
@@ -254,42 +280,31 @@ Type::Type(eSimpleType simple_type) :
 	eType(eSimple),
 	ptr_type(0),
 	simple_type(simple_type),
+	sid(0), // not used for simple types
 	used(false),
 	printed(false),
-        packed_(false)
-{
-	// Nothing else to do.
-}
-
-// --------------------------------------------------------------
- /* copy constructor
-  *******************************************************/
-#if 0
-Type::Type(const Type &t) :
-	eType(t.eType),
-	ptr_type(t.ptr_type),
-	simple_type(t.simple_type),
-	dimensions(t.dimensions),
-	fields(t.fields),
-	used(t.used),
-	printed(t.printed)
+	packed_(false),
+	has_assign_ops_(false),
+	has_implicit_nontrivial_assign_ops_(false)
 {
 	// Nothing else to do.
 }
-#endif
 
 // --------------------------------------------------------------
  /* constructor for struct or union types
   *******************************************************/
 Type::Type(vector<const Type*>& struct_fields, bool isStruct, bool packed,
-			vector<CVQualifiers> &qfers, vector<int> &fields_length) :
+    vector<CVQualifiers> &qfers, vector<int> &fields_length, bool hasAssignOps, bool hasImplicitNontrivialAssignOps) :
     ptr_type(0),
     fields(struct_fields),
     used(false),
     printed(false),
     packed_(packed),
     qfers_(qfers),
-    bitfields_length_(fields_length)
+    bitfields_length_(fields_length),
+	simple_type(MAX_SIMPLE_TYPES),               // not a valid simple type
+	has_assign_ops_(hasAssignOps),
+    has_implicit_nontrivial_assign_ops_(hasImplicitNontrivialAssignOps)
```

Comment:
Can we follow the order in which these members are declared in the Type class, e.g., 

    ptr_type(0),
    simple_type(MAX_SIMPLE_TYPES),               // not a valid simple type
    fields(struct_fields),
    used(false),
    printed(false),
    packed_(packed),
    has_assign_ops_(hasAssignOps),
    has_implicit_nontrivial_assign_ops_(hasImplicitNontrivialAssignOps),
    qfers_(qfers),
    bitfields_length_(fields_length)

Otherwise, GCC/Clang complains about the initialization order.